server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,361 @@
+
+  public RegisteredModelInfo getRegisteredModel(String fullName) {
+    LOGGER.info("Getting registered model: " + fullName);
+    RegisteredModelInfo registeredModelInfo = null;
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        String[] parts = RepositoryUtils.parseFullName(fullName);
+        String catalogName = parts[0];
+        String schemaName = parts[1];
+        String registeredModelName = parts[2];
+        RegisteredModelInfoDAO registeredModelInfoDAO =
+            findRegisteredModel(session, catalogName, schemaName, registeredModelName);
+        if (registeredModelInfoDAO == null) {
+          throw new BaseException(ErrorCode.NOT_FOUND, "Registered model not found: " + fullName);
+        }
+        registeredModelInfo = registeredModelInfoDAO.toRegisteredModelInfo();
+        registeredModelInfo.setCatalogName(catalogName);
+        registeredModelInfo.setSchemaName(schemaName);
+        registeredModelInfo.setFullName(getRegisteredModelFullName(registeredModelInfo));
+        tx.commit();
+        return registeredModelInfo;
+      } catch (Exception e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  private RegisteredModelInfoDAO findRegisteredModel(
+      Session session, String catalogName, String schemaName, String registeredModelName) {
+    UUID schemaId = getSchemaId(session, catalogName, schemaName);
+    return getRegisteredModelDao(session, schemaId, registeredModelName);
+  }
+
+  public RegisteredModelInfo createRegisteredModel(CreateRegisteredModel createRegisteredModel) {
+    ValidationUtils.validateSqlObjectName(createRegisteredModel.getName());
+    long createTime = System.currentTimeMillis();
+    String modelId = UUID.randomUUID().toString();
+    String storageLocation =
+        FileUtils.getModelStorageLocation(
+            createRegisteredModel.getCatalogName(), createRegisteredModel.getSchemaName(), modelId);
+    RegisteredModelInfo registeredModelInfo =
+        new RegisteredModelInfo()
+            .modelId(modelId)
+            .name(createRegisteredModel.getName())
+            .catalogName(createRegisteredModel.getCatalogName())
+            .schemaName(createRegisteredModel.getSchemaName())
+            .comment(createRegisteredModel.getComment())
+            .createdAt(createTime)
+            .updatedAt(createTime);
+    String fullName = getRegisteredModelFullName(registeredModelInfo);
+    registeredModelInfo.setFullName(fullName);
+    LOGGER.info("Creating Registered Model: " + fullName);
+
+    Transaction tx;
+    try (Session session = SESSION_FACTORY.openSession()) {
+      String catalogName = registeredModelInfo.getCatalogName();
+      String schemaName = registeredModelInfo.getSchemaName();
+      UUID schemaId = getSchemaId(session, catalogName, schemaName);
```

Comment:
Shouldn't this be inside the transaction as well ?

Reply:
I guess it depends on how session works.  If multiple sessions can be opened and other transactions can land before we begin our transaction, then yeah, we should move it in the transaction.  I can move it under, thought I think i copied this pattern from one of the other assets (either tables or schemas).

Reply:
What if a delete happens post we get the schema Id. It would be a safer option to include this inside. And follow the same logic elsewhere.

Reply:
Yep.  I'll move it in and see if I can find where else i saw this pattern.

Reply:
Found it: https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/TableRepository.java#L129-L133

Let's cut a separate PR that fixes it for tables.

Reply:
+1